### PR TITLE
Add test to showcase that job-creator would profit from liars

### PR DESCRIPTION
**Pull request by kongzii**

### CODE REVIEW

File: tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    ), "Assumption was that A will receive B's money."
+    assert (
+        ending_balance_B < starting_balance_A
+    ), "Assumption was that B will loose the money he gambled by trying to steal from real job completors."
+
+    print(
+        f"Account A (job creator) ending difference: {ending_balance_A - starting_balance_A}."
+    )
+    print(
+        f"Account B (attacker) ending difference: {ending_balance_B - starting_balance_B}."
```
Comment (kongzii) [inline tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py:215]: Tests in CI are failing atm, but locally:

```
clear && pytest tests_integration_with_local_chain -k test_stealing_on_markets -vvv -s
```

Screenshot by Dropbox Capture

Screenshot by Dropbox Capture

File: tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        web3=local_web3,
+    )
+    print(
+        f"Market created at {created_market.market_event.fixed_product_market_maker_checksummed}"
+    )
+
+    # Initialize OmenMarket and OmenAgentMarket out of it, so we can use it with our standard helper functions.
+    omen_market = OmenMarket.from_created_market(created_market)
+    agent_market = OmenAgentMarket.from_data_model(omen_market)
+    balance_after_market_creation_A, balance_after_market_creation_B = (
```
Comment (evangriffiths) [inline tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py:88]: `balance_after_market_creation_B` unused, so can remove

File: tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    with pytest.raises(Exception) as e_selling:
+        binary_omen_sell_outcome_tx(
+            api_keys_B,
+            sell_yes_for_b,
+            agent_market,
+            binary_outcome=True,
+            auto_withdraw=True,
+            web3=local_web3,
+        )
+    balance_after_failed_trading_A, balance_after_failed_trading_B = (
```
Comment (evangriffiths) [inline tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py:146]: `balance_after_failed_trading_A` unused, can remove

**Review by evangriffiths — COMMENTED** [review]
Nice!

File: tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    account_A, account_B = accounts[0], accounts[1]
+    api_keys_A, api_keys_B = APIKeys(
+        BET_FROM_PRIVATE_KEY=private_key_type(account_A.private_key), SAFE_ADDRESS=None
+    ), APIKeys(
+        BET_FROM_PRIVATE_KEY=private_key_type(account_B.private_key), SAFE_ADDRESS=None
+    )
+    print(f"{api_keys_A.bet_from_address=}, {api_keys_B.bet_from_address=}")
+
+    # Update chain's state with a dummy block.
+    DebuggingContract().inc(api_keys_A, local_web3)
```
[the remark this answers is not in the document]
Comment (kongzii) [reply tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py]: Yeah exactly, explained in detail here https://github.com/gnosis/prediction-market-agent-tooling/pull/437#discussion_r1777337608

I abstracted this away to `mint_new_block` function to make it clear in the code.

File: tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py (hunk 2 above)
Comment (kongzii) [reply tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py:88]: True removed

File: tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py (hunk 3 above)
Comment (kongzii) [reply tests_integration_with_local_chain/markets/omen/test_stealing_on_markets.py:146]: True removed

**Review by evangriffiths — APPROVED** [review]

**Pull request merged**